src/main/java/com/redhat/red/build/koji/KojiClient.java

```diff
@@ -759,10 +810,42 @@ public KojiImportResult importBuild( KojiImport importInfo, Iterable<Supplier<Im
                                                                STANDARD_REQUEST_MODIFIER );
 
             List<KojiArchiveInfo> archives = response.getArchives();
+
             return archives == null ? Collections.emptyList() : archives;
         }, "Failed to retrieve list of artifacts matching archive query: %s", query );
     }
 
+    public List<KojiArchiveInfo> enrichArchiveTypeInfo( List<KojiArchiveInfo> archives, KojiSessionInfo session )
+            throws KojiClientException
+    {
+        Map<Integer, KojiArchiveInfo> archiveIdMap = archives.stream().collect( Collectors.toMap( KojiArchiveInfo::getArchiveId, Function.identity() ) );
+        Map<String, List<KojiArchiveInfo>> buildTypeMap = archives.stream().collect( Collectors.groupingBy( KojiArchiveInfo::getBuildType ) );
+
+        buildTypeMap.forEach( ( buildType, archiveInfos ) -> {
+            List<Object> archiveIds = archiveInfos.stream().map( KojiArchiveInfo::getArchiveId ).collect( Collectors.toList() );
+
+            switch ( buildType )
+            {
+                case "maven":
+                    List<KojiMavenArchiveInfo> mavenArchiveInfos = multiCall( Constants.GET_MAVEN_ARCHIVE, archiveIds, KojiMavenArchiveInfo.class, session );
```

Comment:
Would it be possible to group these by type up front, then make three multicall's into Koji for this information? It might cut down on the comms traffic quite a bit...?

Reply:
> Would it be possible to group these by type up front, then make three multicall's into Koji for this information? It might cut down on the comms traffic quite a bit...?

It's doing that. There's only one call per `buildType`. The `buildTypeMap` is a map where the key is `buildType` and the value is a `List` of all archive ids of that type. The multicall takes the entire list of ids as input.

It is possible to multicall even all three methods as one multicall, but we don't have a good API (yet) for multicalling more than one method at a given time.